processor/memorylimiter/memorylimiter.go

```diff
@@ -142,6 +147,14 @@ func getMemUsageChecker(cfg *Config, logger *zap.Logger) (*memUsageChecker, erro
 	return newPercentageMemUsageChecker(totalMemory, uint64(cfg.MemoryLimitPercentage), uint64(cfg.MemorySpikePercentage))
 }
 
+func (ml *memoryLimiter) start(_ context.Context, _ component.Host) error {
+	ballastSize := ballastextension.GetBallastSize()
+	if ml.ballastSize != ballastSize {
```

Comment:
Is there a reason to have the config in memorylimiter and not just use the ballast extension one directly? Seems more straightforward for users

Reply:
Yes, I like this idea. `ballastSize` in memorylimiter config was added before ballast extension created. Right now, If we want to force `ballastSize` in two components to be matched when ballast memory is set in collector, we better only use `ballastSize` value from one place( suppose to be from `ballast extension`). 

I can delete the `ballastSize` in memorylimiter config and use `ballastSize` value from `ballst extension` in the collector. @bogdandrutu , please let me know what's your thought before I make such changes. Thanks.

Reply:
If we are coupling the functionality in these 2 components this way then I don't see much point in making them separate components. Perhaps just move ballast functionality to memory_limiter and delete the ballast extension?
memory_limiter is virtually a required component anyway. I can't imagine a scenario when one wants to use a ballast because performance matter but they don't want to use a memory limiter.
Thoughts?

Reply:
The concern to move ballast to the processor,
1. Handle the duplicated ballast memory allocations if apply memlimiter processor with ballast enabled to multiple pipelines. 
2. Need to handle the conflicts if multiple mem_limiter instances setup.

> I can't imagine a scenario when one wants to use a ballast because performance matter but they don't want to use a memory limiter.

how about the use case that a user only wants to enable ballast to reduce GCs for better CPU usage but without considering drop any data.

Reply:
I think having the memory limiter take ballast size config from the ballast extension is the way to go.  As @mxiamxia mentions, using an extension for ballast avoids potential issues with multiple allocations or needing to coordinate which memory limiter allocates the ballast.

Reply:
Made the changes to have `ballast extension` be the only place that configure the ballast size value.  `memory_limiter` reads the same value from `ballast extension` for calculating the allocated memory. 

I kept the `ballast_size_mib` config in memory limiter to not bring the breaking change to the existing users and marked it as deprecated.